Userland/Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -333,14 +343,19 @@ Navigable::ChosenNavigable Navigable::choose_a_navigable(StringView name, Tokeni
         chosen = traversable_navigable();
     }
 
-    //  FIXME: 7. Otherwise, if name is not an ASCII case-insensitive match for "_blank",
+    //  7. Otherwise, if name is not an ASCII case-insensitive match for "_blank",
     //     there exists a navigable whose target name is the same as name, currentNavigable's
     //     active browsing context is familiar with that navigable's active browsing context,
     //     and the user agent determines that the two browsing contexts are related enough that
     //     it is ok if they reach each other, set chosen to that navigable. If there are multiple
     //     matching navigables, the user agent should pick one in some arbitrary consistent manner,
     //     such as the most recently opened, most recently focused, or more closely related, and set
     //     chosen to it.
+    else if (same_name_navigable != nullptr && (active_browsing_context()->is_familiar_with(*same_name_navigable->active_browsing_context()))) {
+        // FIXME: Handle multiple name-match case
+        // FIXME: When are these contexts 'not related enough' ?
+        chosen = same_name_navigable;
+    }
```

Comment:
it seems like it would be possible to test this branch even without `window.open()` if we had support for `name` tag in iframes: as I understand `<a href="page.html" target="myIframe">` will navigate `<iframe name="myIframe">`.

Reply:
Hm actually I think that name for iframe and target for HTMLAnchorElement are both in place. This "should work" currently. But I don't think that a test for it will work yet.